services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +45,61 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 10) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  error_code_t errCode;
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  LOG_IF_ERROR_CODE(thermalMgrSendEvent(&event));
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
   while (1) {
-    
+    error_code_t errCode;
+    thermal_mgr_event_t event;
+    BaseType_t eventReceived = xQueueReceive(thermalMgrQueueHandle, &event, 10);
+
+    if (eventReceived == pdTRUE) {
```

Comment:
nitpick: you can just put xQueReceive inside the if and avoid creating a BaseType_t variable